ceph_installer/tests/controllers/test_mon.py

```diff
@@ -40,8 +50,60 @@ def test_configure_missing_fields(self, session):
-        data = dict(host="node1", monitor_interface="eth0", fsid="1720107309134")
+        result = session.app.post_json("/api/mon/configure/", params=self.configure_data)
+        assert result.json['endpoint'] == '/api/mon/configure/'
+        assert result.json['identifier'] is not None
+
+    def test_configure_monitors_not_a_list(self, session, monkeypatch):
+        monkeypatch.setattr(mon.call_ansible, 'apply_async', lambda args, kwargs: None)
+        data = self.configure_data.copy()
+        data["monitors"] = "invalid"
+        result = session.app.post_json("/api/mon/configure/", params=data,
+                                       expect_errors=True)
+        assert result.status_int == 400
+        assert "monitors" in result.json["message"]
+
+    def test_configure_monitors_not_a_list_of_objects(self, session, monkeypatch):
+        monkeypatch.setattr(mon.call_ansible, 'apply_async', lambda args, kwargs: None)
+        data = self.configure_data.copy()
+        data["monitors"] = "['mon1', 'mon2']"
+        result = session.app.post_json("/api/mon/configure/", params=data,
+                                       expect_errors=True)
+        assert result.status_int == 400
+        assert "monitors" in result.json["message"]
+
+    def test_configure_monitors_missing_host_key(self, session, monkeypatch):
+        monkeypatch.setattr(mon.call_ansible, 'apply_async', lambda args, kwargs: None)
+        mons = [{"foo": "bar"}]
+        data = self.configure_data.copy()
+        data['monitors'] = mons
+        result = session.app.post_json("/api/mon/configure/", params=data,
+                                       expect_errors=True)
+        assert result.status_int == 400
+        assert "monitors" in result.json["message"]
+
+    def test_configure_monitors_missing_interface_key(self, session, monkeypatch):
+        monkeypatch.setattr(mon.call_ansible, 'apply_async', lambda args, kwargs: None)
+        mons = [{"host": "mon0.host"}]
+        data = self.configure_data.copy()
+        data['monitors'] = mons
         result = session.app.post_json("/api/mon/configure/", params=data,
                                        expect_errors=True)
+        assert result.status_int == 400
+        assert "monitors" in result.json["message"]
+
+    def test_configure_valid_monitors(self, session, monkeypatch):
+        def check(args, kwargs):
+            inventory = args[0]
+            hosts = inventory[0][1]
+            assert "node1 monitor_interface=eth0" in hosts
+            assert "mon1.host monitor_interface=eth1" in hosts
+
+        monkeypatch.setattr(mon.call_ansible, 'apply_async', check)
+        mons = [{"host": "mon1.host", "interface": "eth1"}]
+        data = self.configure_data.copy()
+        data["monitors"] = mons
+        result = session.app.post_json("/api/mon/configure/", params=data)
+        assert result.status_int == 200
```

Comment:
I don't think you need to be explicit about the 200 here. If the request returns anything else it will error (unless it is returning a different 2XX, which we aren't)